pkg/util/gc_service.go

```diff
@@ -30,10 +33,20 @@ const (
 
 // CheckSafetyOfStartTs checks if the startTs less than the minimum of Service-GC-Ts
 // and this function will update the service GC to startTs
-func CheckSafetyOfStartTs(ctx context.Context, pdCli pd.Client, changefeedID string, startTs uint64) error {
-	minServiceGCTs, err := pdCli.UpdateServiceGCSafePoint(ctx, cdcChangefeedCreatingServiceGCSafePointID+changefeedID,
-		cdcChangefeedCreatingServiceGCSafePointTTL, startTs)
-	if err != nil {
+func CheckSafetyOfStartTs(ctx context.Context, pdCli pd.Client, changefeedID string, startTs uint64) (err error) {
+	var minServiceGCTs uint64
+	// pd leader switch may happen, so just retry it.
+	if err := retry.Do(ctx, func() error {
+		minServiceGCTs, err = pdCli.UpdateServiceGCSafePoint(ctx, cdcChangefeedCreatingServiceGCSafePointID+changefeedID,
```

Comment:
Could you add a test? You can use mock pd client to simulate error.

https://github.com/pingcap/ticdc/blob/596efd4b5fc05188a4937564266683aa7f2b6d8b/cdc/kv/client_mock_test.go#L28-L33